### DISCUSSION TITLE
How does react-hook-form set the input default value?

### DISCUSSION THREAD

**Discussion by raigoinabox** [react-hook-form/react-hook-form#10822] [Q&A] [answered]
I'm trying to extend react-hook-form for my own project. My approach is to wrap the useForm call with my own hook and input modified values the real hook and output modified functions with extra functionality to my code. I'm trying to add a transform property to the register call so that values can be formatted before supplying them into the input (so I can format monetary values or dates and keep a standard data format for "outside components").

I have figured something out for the setValue call by wrapping it with my own, but I don't understand how react-hook-form sets the input initial values. It doesn't seem like that the register call returns defaultValue property or something like it, which makes the most sense to me. Does react-hook-form does some post-value setting with the input ref in an useEffect hook? If so, can I somehow access it and modify it so that it goes through my supplied function?

**Comment by Moshyfawn** [comment, marked as answer]
The default values is set via element `ref`. You can see the code related to it [here](https://github.com/react-hook-form/react-hook-form/blob/5f3df1227c381a9127ab27f7d1d98a9cefee33d5/src/logic/createFormControl.ts#L979C14-L979C14) and follow it for more information
